iocage_lib/ioc_fetch.py

```diff
@@ -264,19 +262,23 @@ def fetch_release(self, _list=False):
                     _callback=self.callback,
                     silent=self.silent)
 
-            dataset = f"{self.iocroot}/download/{self.release}"
-            pool_dataset = f"{self.pool}/iocage/download/{self.release}"
+            pool_dataset_name = f"{self.pool}/iocage/download/{self.release}"
 
-            if os.path.isdir(dataset):
+            try:
+                dataset = self.zfs.get_dataset(
+                    f"{self.iocroot}/download/{self.release}"
+                )
+                return
```

Comment:
This changes the behavior of fetch. We check if it's a directory to see if it's mounted (that can be improved), but returning here means the fetch process will not re-extract and update the RELEASE.